tests/operator_test.go

```diff
@@ -2213,6 +2214,49 @@ spec:
 			errMsg := fmt.Sprintf("Unsupported value: \"%s\"", incorrectOperator)
 			patchKvWorkloads(&incorrectWorkload, true, errMsg)
 		})
+
+		It("[test_id:5760]should check if vm with windows nodeselector is in scheduling phase", func() {
+			vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+			vmi.Spec.NodeSelector = map[string]string{k8sv1.LabelOSStable: "windows"}
+			vmi, err = virtClient.VirtualMachineInstance(util2.NamespaceTestDefault).Create(vmi)
```

Comment:
would you use the more compact `tests.RunVMIAndExpectScheduling` ?

Reply:
RunVMIAndExpectScheduling waits for vm to be in running state. In this test, vm will never run, because it tries to be scheduled on windows node.

Reply:
> `RunVMIAndExpectScheduling` waits for vm to be in running state.

This is very surprising; I find the name misleading. What's the difference between this function and `RunVMIAndExpectLaunch` then?

Reply:
I was confused by the params which is passed to a function inside RunVMIAndExpectScheduling function. The function is looking for 3 different phases (Scheduling, Scheduled, Running) and if vm is in one of these phases. I refactor my code to use this function.

Reply:
Thanks. But please consider my bigger question - should this be an e2e test?

Reply:
After discussion with @sarahbx all of these test should be e2e